src/components/Profile/ProfileForm.tsx

```diff
@@ -0,0 +1,124 @@
+import {useRouter} from 'next/router'
+import {FC} from 'react'
+import {SubmitHandler, useForm} from 'react-hook-form'
+
+import styles from '@/components/FormItems/Form.module.scss'
+import {FormInput} from '@/components/FormItems/FormInput/FormInput'
+import {SelectOption} from '@/components/FormItems/FormSelect/FormSelect'
+import {IGeneralPostResponse} from '@/types/api/general'
+import {Profile} from '@/types/api/personal'
+import {AuthContainer} from '@/utils/AuthContainer'
+import {useSeminarInfo} from '@/utils/useSeminarInfo'
+
+import {Button} from '../Clickable/Clickable'
+import {SchoolSubForm} from '../SchoolSubForm/SchoolSubForm'
+
+export type ProfileFormValues = {
+  first_name?: string
+  last_name?: string
+  phone?: string
+  parent_phone?: string
+  new_school_description?: string
+  without_school: boolean
+  school?: SelectOption | null
+  school_not_found: boolean
+  grade: number | ''
+}
+
+const defaultValues: ProfileFormValues = {
+  first_name: '',
+  last_name: '',
+  phone: '',
+  parent_phone: '',
+  new_school_description: '',
+  without_school: false,
+  school: null,
+  school_not_found: false,
+  grade: '',
+}
+
+export const ProfileForm: FC = () => {
+  const {isAuthed} = AuthContainer.useContainer()
+
+  const {data} = useQuery({
+    queryKey: ['personal', 'profiles', 'myprofile'],
+    queryFn: () => axios.get<Profile>(`/api/personal/profiles/myprofile`),
+    enabled: isAuthed,
+  })
+  const profile = data?.data
+  const profileValues: ProfileFormValues = {
+    first_name: profile?.first_name,
+    last_name: profile?.last_name,
+    phone: profile?.phone ?? '',
+    parent_phone: profile?.parent_phone ?? '',
+    new_school_description: '',
+    without_school: profile?.school_id === 1,
+    school: ({id: profile?.school.code, label: profile?.school.verbose_name} as SelectOption) ?? null,
+    school_not_found: profile?.school_id === 0,
+    grade: profile?.grade ?? '',
+  }
+
+  const {handleSubmit, control, watch, setValue} = useForm<ProfileFormValues>({
+    defaultValues,
+    values: profileValues,
+  })
+
+  const scrollToTop = () => {
+    window.scrollTo({
+      top: 0,
+      behavior: 'smooth',
+    })
+  }
+
+  const router = useRouter()
+
+  const {seminar} = useSeminarInfo()
+
+  const transformFormData = (data: ProfileFormValues) => ({
+    profile: {
+      first_name: data.first_name,
+      last_name: data.last_name,
+      school: data.school?.id,
+      phone: data.phone,
+      parent_phone: data.parent_phone,
+      grade: data.grade,
+    },
+    new_school_description: data.new_school_description || '',
+  })
+
+  const {mutate: submitFormData} = useMutation({
+    mutationFn: (data: ProfileFormValues) => {
+      return axios.put<IGeneralPostResponse>(`/api/user/user`, transformFormData(data))
+    },
+    onSuccess: () => router.push(`/${seminar}/profil`),
+  })
+
+  const onSubmit: SubmitHandler<ProfileFormValues> = async (data) => {
+    submitFormData(data)
+  }
```

Comment:
asi raz pridame daky eslint rule co trochu zjednoti/da bacha na async funkcie... tu deklarujeme fuknciu ako async ale nevolame v nej await, co nejakemu rulu vadi, ale v podstate je to tu v pohode